### DISCUSSION TITLE
`auth.css` is fetched with HTTP instead of HTTPS when running behind a reverse proxy

### DISCUSSION THREAD

**Discussion by djpetti** [fief-dev/fief#297] [Q&A]
I'm aware of #178, and I tried setting the `FORWARDED_ALLOWED_IPS` variable as specified, but it doesn't fix the problem.  I'm still seeing this behavior. My nginx configuration looks like this:

```
proxy_set_header  X-Real-IP  $remote_addr;
proxy_set_header  X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header  X-Forwarded-Host $http_host;
proxy_set_header  Host $http_host;

upstream auth {
    server  auth_server:8000;
}

server {
    server_name _;
    listen 8099 ssl http2 default_server;
    listen [::]:8099 ssl http2 default_server;
    ssl_certificate /etc/ssl/certs/nginx-selfsigned.crt;
    ssl_certificate_key /etc/ssl/private/nginx-selfsigned.key;

    location / {
        proxy_pass http://auth/;
    }
}
```

My Fief environment variables are set like so:
```
SECRET=<redacted>
FIEF_CLIENT_ID=<redacted>
FIEF_CLIENT_SECRET=<redacted>
ENCRYPTION_KEY=<redacted>
PORT=8000
ROOT_DOMAIN=localhost:8099
FIEF_DOMAIN=localhost:8099
FIEF_MAIN_USER_EMAIL=djpetti@gmail.com
FIEF_MAIN_USER_PASSWORD=<redacted>

# Database settings
DATABASE_TYPE=MYSQL
DATABASE_HOST=metadata_store
DATABASE_PORT=3306
DATABASE_USERNAME=root
DATABASE_PASSWORD=<redacted>
DATABASE_NAME=mallard_meta

REDIS_URL=redis://auth_redis:6379

# Enable HTTPS behind proxy.
FORWARDED_ALLOW_IPS=*

LOG_LEVEL="DEBUG"
```

Am I missing something here? I would appreciate if someone could post a working Nginx configuration for this.

**Comment by fief-bailiff** [comment]
Hail, @djpetti 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment]
`FORWARDED_ALLOWED_IPS` is indeed essential in the context of a reverse proxy.

I think however you may need to pass the `X-Forwarded-Proto` header to make it work with SSL. You probably can draw inspiration from this Uvicorn guide: https://www.uvicorn.org/deployment/#running-behind-nginx
